bot/src/index.ts

```diff
@@ -18,7 +18,7 @@ const loop = async function (): Promise<void> {
         }
         const newAuctions = getNewAuctionsFromActiveAuctions(activeAuctions);
         newAuctions.map(notify);
-        activeAuctions.map(participate);
+        await participate(activeAuctions);
```

Comment:
> Does this fix the issue?

Now it's possible that `participate` will be called another time before it's previous execution completes. But I think we're already insured against it, right?

Reply:
> Now it's possible that `participate` will be called another time before it's previous execution completes. But I think we're already insured against it, right?

We have a system in place to prevent this! https://github.com/sidestream-tech/unified-auctions-ui/blob/main/bot/src/keeper.ts#L13